code/game/machinery/newscaster.dm

```diff
@@ -693,6 +756,16 @@ var/list/obj/machinery/newscaster/allCasters = list() //Global list that will co
 
 		photo_data = new(selection, 1)
 
+/obj/machinery/newscaster/proc/AttachPaper(mob/user as mob)
+	if(paper_data || paper_name)
+		paper_name = ""
+		paper_data = ""
+	
+	if(istype(user.get_active_hand(), /obj/item/weapon/paper))
```

Comment:
Failure message as an `else` condition here might be a nice touch?

Reply:
Done.